### ISSUE TITLE
Supervisors download all task documents from CHWs they supervise

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#6191]
**Describe the bug**
As part of Rules-Engine v2, tasks are now stored to disk and replicated up and down.
However, a stored task has this structure:

```
{
    "type": "task",
    "authoredOn": "<timestamp>",
    "state": "<some_task_state>"
    "stateHistory": [{ "state": "<some_task_state>", "timestamp": "<timestamp>" }, ...],
    "user": "<users's contact document id>",
    "requester": "<taskEmission.doc.contact._id>",
    "owner": "<taskEmission.contact._id>",
    "emission": { ... emission data }
}
```

relevant code here: https://github.com/medic/cht-core/blob/master/shared-libs/rules-engine/src/transform-task-emission-to-doc.js#L26

The `user` field is used for determining replication permissions for the doc and is the value emitted in `medic/docs_by_replication_key`: https://github.com/medic/cht-core/blob/master/ddocs/medic/views/docs_by_replication_key/map.js#L71 .
Because the field contains the uuid of the user's contact doc, supervisors are most likely allowed to view the people they supervise, hence they will download all the tasks of these users.

**To Reproduce**

Using "legacy" hierarchy:
1. Create a CHW under a `health_center`, add some families and people to the families and add at least one report that would generate a task. Sync!
2. Create a Supervisor above the CHW, in `district_hospital` level and give it `replication_depth` = 2. This means that he will download the `health_center`, the CHW Contact document and the families, but none of the people in the families or their reports.
3. Log in with the Supervisor account and check the tasks tab. Notice that you see the same task as the "CHW".

**Expected behavior**
I believe it is intended for supervisors not to download supervisee tasks, as that could prove severely detrimental for their devices performance.

**Environment**
- App: webapp
- Version: 3.8.x

**Additional context**
It appears that simply switching the `task.user` field to contain the actual user-settings document id (`org:couchdb:user:<username>`) would solve the replication issue, as only the users themselves have access to the user-settings docs. 

If, in the future, we want to have the ability to generate tasks for other users, not based on their username, we could additionally have `medic/docs_by_replication_key` emit an additional value that would send the task to the correct people - for example emit the `owner` and have the task downloaded to everyone who sees the person.

**Comment by kennsippell** [comment]
Ready for AT on `6191-no-supervisee-tasks`

**Comment by dianabarsan** [comment]
When AT-ing, please make sure to check all offline users' replication correctness. 
A new replication "rule" was added so we would have target documents be downloaded only to the user generating them.

**Comment by ngaruko** [comment]
LGTM.
Will focus on this during RT with more hierarchies

**Comment by ngaruko** [comment]
@kennsippell Can we please back port this to 3.8 (and eventually any other recently merged PR that's needed for 3.8)?
